common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -2591,6 +2591,10 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
     HIVE_CTE_MATERIALIZE_THRESHOLD("hive.optimize.cte.materialize.threshold", -1,
```

Comment:
Should we set the default to `2` now that it is only triggered in very specific cases?

Reply:
done